src/List/List.js

```diff
@@ -25,86 +25,53 @@ const ListContext = createContext({
 });
 ListContext.displayName = 'ListContext';
 
-class List extends Component {
-  constructor(props) {
-    super(props);
-
-    const { nested, open, minimal, multiSelect, selectable } = this.props;
-
-    this.state = {
-      nested,
-      open,
-      minimal,
-      multiSelect,
-      selectable
-    };
-
-    this.listId = this.props.id || uniqid();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { nested, open, minimal, multiSelect, selectable } = this.props;
-
-    if (
-      prevProps.nested !== nested ||
-      prevProps.open !== open ||
-      prevProps.minimal !== minimal ||
-      prevProps.multiSelect !== multiSelect ||
-      prevProps.selectable !== selectable
-    ) {
-      this.setState({
-        nested,
-        open,
-        minimal,
-        multiSelect,
-        selectable
-      });
-    }
+const List = ({
+  children,
+  nested,
+  open,
+  minimal,
+  multiSelect,
+  selectable,
+  ...other
+}) => {
+  const listContext = useContextState({
+    nested,
+    open,
+    minimal,
+    multiSelect,
+    selectable
+  });
+  const listEl = useRef(null);
+
+  let listMaxHeight = 'none';
+  if (open === false) {
+    listMaxHeight = '0px';
+  } else if (listEl.current && nested) {
+    listMaxHeight = 0;
+    listEl.current.childNodes.forEach(child => {
+      listMaxHeight = listMaxHeight + child.getBoundingClientRect().height;
+    });
+    listMaxHeight = Math.ceil(listMaxHeight) + 'px';
```

Comment:
This would probably be a good use case for `reduce`

Reply:
`childNodes` is a `NodeList` which doesn't have `reduce`